nucypher_cli/demos/finnegans-wake-demo.py

```diff
@@ -3,55 +3,40 @@
 
 # WIP w/ hendrix@3.0.0
 
-import binascii
 import datetime
-import logging
 import sys
 
 import maya
+from umbral.keys import UmbralPublicKey
 
+from nucypher.blockchain.eth.chains import Blockchain
 from nucypher.characters import Alice, Bob, Ursula
 from nucypher.data_sources import DataSource
-# This is already running in another process.
-from nucypher.network.middleware import RestMiddleware
-from umbral.keys import UmbralPublicKey
 
-root = logging.getLogger()
-root.setLevel(logging.DEBUG)
 
-ch = logging.StreamHandler(sys.stdout)
-ch.setLevel(logging.INFO)
-formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
-ch.setFormatter(formatter)
-root.addHandler(ch)
+##############################################
+# This is already running in another process.
+##############################################
 
-teacher_dht_port = sys.argv[2]
-teacher_rest_port = int(teacher_dht_port) + 100
-with open("examples-runtime-cruft/node-metadata-{}".format(teacher_rest_port), "r") as f:
-    f.seek(0)
-    teacher_bytes = binascii.unhexlify(f.read())
-URSULA = Ursula.from_bytes(teacher_bytes, federated_only=True)
-print("Will learn from {}".format(URSULA))
+BLOCKCHAIN = Blockchain.connect()
 
-# network_middleware = SandboxRestMiddleware([URSULA])
+URSULA = Ursula.from_config()
 
 #########
 # Alice #
 #########
 
-ALICE = Alice(network_middleware=RestMiddleware(),
-              known_nodes=(URSULA,),  # in lieu of seed nodes
-              federated_only=True,
-              always_be_learning=True)  # TODO: 289
+ALICE = Alice.from_config()
 
 # Here are our Policy details.
-policy_end_datetime = maya.now() + datetime.timedelta(days=5)
+policy_end_datetime = maya.now() + datetime.timedelta(days=201)
 m = 2
 n = 3
 label = b"secret/files/and/stuff"
 
 # Alice grants to Bob.
-BOB = Bob(known_nodes=(URSULA,), federated_only=True, always_be_learning=True)
+BOB = Bob.from_config()
```

Comment:
Jeez, that's... I dunno it's disappointing to me to see so much live conceptualization pushed down into config, especially in a demo that is designed for the reader first.

Reply:
Hey - Feel free to use it either way - the `from_config` methods allow overrides by passing params here.

Reply:
Agreed - Let's do some in-person review.

Reply:
I think I agree with @jMyles.

Reply:
#393